OptimoveSDK/Sources/Classes/PreferenceCenter/OptimovePreferenceCenter.swift

```diff
@@ -0,0 +1,197 @@
+import Foundation
+import OptimoveCore
+
+@available(iOS 13.0, *)
+public class OptimovePreferenceCenter {
+    enum Error: LocalizedError {
+        case alreadyInitialized
+        case notInitialized
+
+        var errorDescription: String? {
+            switch self {
+            case .alreadyInitialized:
+                return "The PreferenceCenterSDK has already been initialized."
+            case .notInitialized:
+                return "Preference center has not been initialized."
+            }
+        }
+    }
+
+    private static var instance: OptimovePreferenceCenter?
+    private var networkClient: NetworkClient?
+    private var storage: OptimoveStorage?
+
+    static var isSdkRunning: Bool {
+        return Optimove.getConfig()?.getPreferenceCenterConfig() != nil
+    }
+
+    public enum ResultType {
+        case success
+        case errorUserNotSet
+        case errorNotConfigured
+        case error
+    }
+
+    public typealias PreferencesGetHandler = (_ result: ResultType, _ preferences: Preferences?) -> Void
+    public typealias PreferencesSetHandler = (_ result: ResultType) -> Void
+
+    public static func getInstance() throws -> OptimovePreferenceCenter {
+        guard let instance = instance else {
+            throw Error.notInitialized
+        }
+        return instance
+    }
+
+    static func initialize(storage: OptimoveStorage, networkClient: NetworkClient) throws {
+        if let config = Optimove.getConfig() {
+            if instance !== nil && config.features.contains(.delayedConfiguration) {
```

Comment:
this actually might be redundant because we have this check in `OptimoveConfig`

```swift
    let preferenceCenterConfig: PreferenceCenterConfig? = {
            guard features.contains(.preferenceCenter) else {
                if preferenceCenterCredentials != nil {
                    Logger.error("Preference center credentials provided, but the feature was not requested.")
                }
                return nil
            }
            
              // referring to this
            if preferenceCenterCredentials == nil {
                if !features.contains(.delayedConfiguration) {
                    Logger.error("Preference center could not be initialized due to missing credentials.")
                }
                return nil
            }

            return getPreferenceCenterConfig(from: preferenceCenterCredentials)
        }()

```

Reply:
it might be redundant because of this

```
    func isOptimobileConfigured() -> Bool {
        return features.contains(.optimobile)
    }
    
    func isPreferenceCenterConfigured() -> Bool {
        return features.contains(.preferenceCenter) && preferenceCenterConfig != nil
    }

```

because of ^ it's not going to be initialised. But ^ is not entirely correct -- `isXConfigured` checks if we want it in general, and if we do  for other modules it can start doing work before credentials given. 

So, a module should be initialised when it is configured, and it is configured when it is among features. And then when credentials given it would be initialised again

lol, this is confusing :) but i think if we try to keep modules behaving more similar, will be slightly simpler to reason about